src/main/java/uk/gov/pay/directdebit/gatewayaccounts/api/CreateGatewayAccountRequest.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.pay.directdebit.gatewayaccounts.api;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.hibernate.validator.constraints.NotEmpty;
+import uk.gov.pay.directdebit.gatewayaccounts.model.GatewayAccount.Type;
+import uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider;
+import uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProviderAccessToken;
+import uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProviderOrganisationIdentifier;
+
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CreateGatewayAccountRequest {
+
+    @NotNull
+    private final PaymentProvider paymentProvider;
+
+    @NotEmpty
+    @Size(min = 1, max = 50, message = "service_name must be between {min} and {max} characters.")
+    private final String serviceName;
+
+    @NotNull
+    private final Type type;
+
+    private final String description;
```

Comment:
this looks great, much cleaner. It would be good (maybe in separate PR) to tighten up validation here - all these params must have a max length, and it would be good to catch them here rather than later

Reply:
yes good idea. I've added for `description` and `analysticsId` which are Strings stored into `VARCHAR(255)`. And in this new format, it was super easy to do :)

Reply:
🎉